features/FEATURE_LWIP/lwip-interface/lwipopts.h

```diff
@@ -102,10 +102,17 @@
 #define MBED_CONF_LWIP_DEFAULT_THREAD_STACKSIZE    512
 #endif
 
+// Thread stack size for private PPP thread
+#ifndef MBED_CONF_LWIP_PPP_THREAD_STACKSIZE
+#define MBED_CONF_LWIP_PPP_THREAD_STACKSIZE    300
+#endif
+
 #if LWIP_DEBUG
 #define DEFAULT_THREAD_STACKSIZE    MBED_CONF_LWIP_DEFAULT_THREAD_STACKSIZE*2
+#define PPP_THREAD_STACK_SIZE       MBED_CONF_LWIP_PPP_THREAD_STACKSIZE*4
 #else
 #define DEFAULT_THREAD_STACKSIZE    MBED_CONF_LWIP_DEFAULT_THREAD_STACKSIZE
```

Comment:
I think that PPP thread size define has been misplaced  - lwipopts.h is supposed to be configuring lwIP itself.

The PPP thread is outside of lwIP - there needs to be a data pump thread to feed data in. It can't share lwIP's internal thread as it's not exposed.

The data pump is a very simple single event (make non-blocking FileHandle::read, call lwIP) being deferred from interrupt context (sigio) - it's a textbook example of something that would use a shared event queue.